src/utils/ucx/config.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "config.h"
+
+#include <stdexcept>
+
+#include <absl/strings/str_format.h>
+
+#include "common/nixl_log.h"
+
+namespace nixl::ucx {
+void
+config::modify (std::string_view key, std::string_view value) const {
+    const char *env_val = std::getenv (absl::StrFormat ("UCX_%s", key.data()).c_str());
+    if (env_val) {
+        NIXL_DEBUG << "UCX env var has already been set: " << key << "=" << env_val;
+    } else {
+        modifyUcpConfig (key, value);
+    }
+}
+
+void
+config::modifyAlways (std::string_view key, std::string_view value) const {
+    modifyUcpConfig (key, value);
```

Comment:
`modifyUcpConfig` looks redundant, `modifyAlways` can be called after `env_val` check in `modify`